Index: Reconstruction/AODFix/python/AODFix_r207.py
===================================================================
--- Reconstruction/AODFix/python/AODFix_r207.py	(revision 788049)
+++ Reconstruction/AODFix/python/AODFix_r207.py	(working copy)
@@ -756,7 +756,7 @@
                                                        TaggerList = ['IP3D','MV2c00','MV2c10',
                                                                      'MV2c20','MV2c100','MV2m',
                                                                      'MV2c10hp','MV2cl100',
-                                                                     'MVb','JetVertexCharge'])
+                                                                     'MVb'])
                 jet = jet.replace("Track", "PV0Track")
                 SAbtagger = StandAloneJetBTaggerAlg(name=SA + AuthorSubString[i].lower(),
                                           JetBTaggerTool=btagger,
