Index: PhysicsAnalysis/JetTagging/JetTagAlgs/BTagging/python/BTaggingConfiguration_LoadTools.py
===================================================================
--- PhysicsAnalysis/JetTagging/JetTagAlgs/BTagging/python/BTaggingConfiguration_LoadTools.py	(revision 788054)
+++ PhysicsAnalysis/JetTagging/JetTagAlgs/BTagging/python/BTaggingConfiguration_LoadTools.py	(working copy)
@@ -327,6 +327,32 @@
                                                  'parRCOM': coneCoM}) # default: 0.8
 
     defaultBTagTrackToJetAssociator = associatorCoMName
+  # for RNN studies
+  elif ("RNNCone" in JetCollection) and ("ExKt2Sub" not in JetCollection) and ("ExKt3Sub" not in JetCollection):   # parent jet, but not subjets
+    if "RNNCone06" in JetCollection:
+      associatorRNNName = "BTagTrackToJetAssociatorRNN06"
+      coneRNN = 0.6
+    elif "RNNCone07" in JetCollection:
+      associatorRNNName = "BTagTrackToJetAssociatorRNN07"
+      coneRNN = 0.7
+    elif "RNNCone08" in JetCollection:
+      associatorRNNName = "BTagTrackToJetAssociatorRNN08"
+      coneRNN = 0.8
+    elif "RNNCone10" in JetCollection:
+      associatorRNNName = "BTagTrackToJetAssociatorRNN10"
+      coneRNN = 1.0
+    else:
+      associatorRNNName = "BTagTrackToJetAssociatorRNN10"
+      coneRNN = 1.0
+
+    ConfInstance.setupTrackAssociator(associatorRNNName, JetCollection, ToolSvc,
+                                      Verbose = BTaggingFlags.OutputLevel < 3,
+                                      options = {
+                                                  "trackCone": coneRNN,
+                                                  "useVariableSizedTrackCone": False,
+                                                })
+
+    defaultBTagTrackToJetAssociator = associatorRNNName
   else:
     defaultBTagTrackToJetAssociator = "BTagTrackToJetAssociator"
 
