Index: PhysicsAnalysis/DerivationFramework/DerivationFrameworkFlavourTag/python/HbbCommon.py
===================================================================
--- PhysicsAnalysis/DerivationFramework/DerivationFrameworkFlavourTag/python/HbbCommon.py	(revision 788054)
+++ PhysicsAnalysis/DerivationFramework/DerivationFrameworkFlavourTag/python/HbbCommon.py	(working copy)
@@ -50,7 +50,7 @@
       JetAlgorithm = algj,
       JetRadius = 10.0,
       PtMin = 1000,
-      ExclusiveNJets = 2,
+      ExclusiveNJets = nsubjet,
       doTrack = doTrackSubJet,
       InputJetContainerName = JetCollectionName,
       SubjetContainerName = SubjetContainerName,
@@ -70,17 +70,17 @@
 #===================================================================
 # Build ExKt Subjets
 #===================================================================
-def addExKt(FTAG5Seq, ToolSvc, ExKtJetCollection__FatJet, doTrackSubJet):
+def addExKt(FTAG5Seq, ToolSvc, ExKtJetCollection__FatJet, nSubJet, doTrackSubJet):
     ExKtJetCollection__SubJet = []
     for JetCollectionExKt in ExKtJetCollection__FatJet:
         # build ExKtbbTagTool instance
-        (ExKtbbTagToolInstance, SubjetContainerName) = buildExclusiveSubjets(ToolSvc, JetCollectionExKt, "Kt", 2, doTrackSubJet)
+        (ExKtbbTagToolInstance, SubjetContainerName) = buildExclusiveSubjets(ToolSvc, JetCollectionExKt, "Kt", nSubJet, doTrackSubJet)
         ExKtJetCollection__SubJet += [SubjetContainerName]
 
         # build subjet collection through JetRecTool
         from JetRec.JetRecConf import JetRecTool
         jetrec = JetRecTool(
-                             name = "JetRecTool_ExKtbb_%s" % (JetCollectionExKt),
+                             name = "JetRecTool_ExKtbb_"+SubjetContainerName,
                              OutputContainer = JetCollectionExKt,
                              InputContainer = JetCollectionExKt,
                              JetModifiers = [ExKtbbTagToolInstance],
@@ -88,7 +88,7 @@
 
         ToolSvc += jetrec
         FTAG5Seq += JetAlgorithm(
-                                 name = "JetAlgorithm_ExKtbb_%s" % (JetCollectionExKt),
+                                 name = "JetAlgorithm_ExKtbb_%s_%s" % (JetCollectionExKt, SubjetContainerName),
                                  Tools = [jetrec],
                                 )
     return ExKtJetCollection__SubJet
@@ -96,15 +96,15 @@
 #===================================================================
 # Build CoM Subjets
 #===================================================================
-def addExCoM(FTAG5Seq, ToolSvc, ExKtJetCollection__FatJet, doTrackSubJet):
+def addExCoM(FTAG5Seq, ToolSvc, ExKtJetCollection__FatJet, nSubJet, doTrackSubJet):
     ExCoMJetCollection__SubJet = []
     for JetCollectionExCoM in ExKtJetCollection__FatJet:
-        (ExCoMbbTagToolInstance, SubjetContainerName) = buildExclusiveSubjets(ToolSvc, JetCollectionExCoM, "CoM", 2, doTrackSubJet)
+        (ExCoMbbTagToolInstance, SubjetContainerName) = buildExclusiveSubjets(ToolSvc, JetCollectionExCoM, "CoM", nSubJet, doTrackSubJet)
         ExCoMJetCollection__SubJet += [SubjetContainerName]
 
         from JetRec.JetRecConf import JetRecTool
         jetrec = JetRecTool(
-                             name = "JetRecTool_ExCoMbb_%s" % (JetCollectionExCoM),
+                             name = "JetRecTool_ExCoMbb_"+SubjetContainerName,
                              OutputContainer = JetCollectionExCoM,
                              InputContainer = JetCollectionExCoM,
                              JetModifiers = [ExCoMbbTagToolInstance],
@@ -112,7 +112,7 @@
 
         ToolSvc += jetrec
         FTAG5Seq += JetAlgorithm(
-                                 name = "JetAlgorithm_ExCoMbb_%s" % (JetCollectionExCoM),
+                                 name = "JetAlgorithm_ExCoMbb_%s_%s" % (JetCollectionExCoM, SubjetContainerName),
                                  Tools = [jetrec],
                                 )
     return ExCoMJetCollection__SubJet
